modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/SqlQueryProcessor.java

```diff
@@ -366,7 +370,7 @@ public CompletableFuture<Void> closeAsync() {
                     );
                 });
 
-        stage.whenComplete((cur, ex) -> {
+        stage = stage.whenComplete((cur, ex) -> {
```

Comment:
the idea behind is that you return future with attached cancellation handler to the client. Now you return the future of the handler to the client, and this doesn't make any sense